examples/v2/chat2.nim

```diff
@@ -115,6 +116,38 @@ proc showChatPrompt(c: Chat) =
     stdout.flushFile()
     c.prompt = true
 
+proc printReceivedMessage(c: Chat, msg: WakuMessage) =
+  when PayloadV1:
+      # Use Waku v1 payload encoding/encryption
+      let
+        keyInfo = KeyInfo(kind: Symmetric, symKey: c.symKey)
+        decodedPayload = decodePayload(decoded.get(), keyInfo)
+
+      if decodedPayload.isOK():
+        let
+          pb = Chat2Message.init(decodedPayload.get().payload)
+          chatLine = if pb.isOk: pb[].toString()
+                    else: string.fromBytes(decodedPayload.get().payload)
+        echo &"{chatLine}"
+        c.prompt = false
+        showChatPrompt(c)
+        trace "Printing message", topic=DefaultTopic, chatLine,
+          contentTopic = msg.contentTopic
+      else:
+        debug "Invalid encoded WakuMessage payload",
+          error = decodedPayload.error
+  else:
+    # No payload encoding/encryption from Waku
+    let
+      pb = Chat2Message.init(msg.payload)
+      chatLine = if pb.isOk: pb[].toString()
+                else: string.fromBytes(msg.payload)
```

Comment:
wondering why the message is shown despite its conversion to a chat2message is failed?

Reply:
Initially, this was to be compatible with previous versions of `chat2` that simply passed around the message payload without encoding into `Chat2Message`. For now, it will at least allow those using the testnet content topic to pass around non-specified payloads if they so choose. Since the testnet is designed for experimental use, I thought this a better approach than to quietly drop the message. May help with debugging different clients! 😄